### PR TITLE
Adds product property assoc feature

### PR DESCRIPTION
**Pull request by anantanant2015**
Adds product property assoc feature

## Motivation and Context

- This would enable feature of product association with existing properties in
  the system.

## Describe your changes

- Adds routes for product property assoc.
- Adds CRUD operations in product controller for product
  property assoc.
- Adds templates for product property assoc.
- Updates view for product property assoc.
- Adds model, schema for product property.

## Any further comments?

------

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

------
Dear Gatekeeper,

Please make sure that the commits that will be merged or rebased into the base
branch are [formatted according to our standards][commit-style].

> The only additional requirement is that the the PR number must appear in the
> commit title in brackets: `(#XXX)`

[commit-style]: https://github.com/aviabird/snitch/blob/develop/CONTRIBUTING.md#styling

### CODE REVIEW

**Comment by ashish173** [comment]
@arjun289 may I also request you to change any wrong labels in the stock section and the image section. @anantanant2015 and I missed it when we fixed things day before yesterday. Thanks!

File: apps/admin_app/lib/admin_app_web/controllers/product_controller.ex [hunk 1]
```diff
@@ -238,4 +249,63 @@ defmodule AdminAppWeb.ProductController do
...
+      changeset: changeset,
+      product_id: params["product_id"]
+    )
+  end
+
+  def create_property(conn, params) do
+    with %ProductSchema{} = _product <- ProductModel.get(params["product_id"]),
+         %Property{} = _property <- Model.Property.get(params["product_property"]["property_id"]),
+         {:ok, _product_property} <- Model.ProductProperty.create(params["product_property"]) do
+      redirect(conn, to: "/products/#{params["product_id"]}/edit")
```
Comment (SagarKarwande) [inline apps/admin_app/lib/admin_app_web/controllers/product_controller.ex]: Use path helpers

File: apps/admin_app/lib/admin_app_web/controllers/product_controller.ex [hunk 2]
```diff
@@ -238,4 +249,63 @@ defmodule AdminAppWeb.ProductController do
...
+  end
+
+  def update_property(conn, params) do
+    with %ProductProperty{} = product_property <-
+           Model.ProductProperty.get_by(%{
+             product_id: params["product_property"]["product_id"],
+             property_id: params["product_property"]["property_id"]
+           }),
+         {:ok, _} = Model.ProductProperty.update(product_property, params["product_property"]) do
+      redirect(conn, to: "/products/#{params["product_property"]["product_id"]}/edit")
```
Comment (SagarKarwande) [inline apps/admin_app/lib/admin_app_web/controllers/product_controller.ex]: use path helpers

File: apps/admin_app/lib/admin_app_web/controllers/product_controller.ex [hunk 3]
```diff
@@ -238,4 +249,63 @@ defmodule AdminAppWeb.ProductController do
...
+    with %ProductProperty{} = product_property <-
+           Model.ProductProperty.get_by(%{
+             product_id: params["product_id"],
+             property_id: params["property_id"]
+           }),
+         {:ok, _} <- Model.ProductProperty.delete(product_property) do
+      conn
+      |> put_flash(:info, "Product property deleted successfully")
+
+      redirect(conn, to: "/products/#{params["product_id"]}/edit")
```
Comment (SagarKarwande) [inline apps/admin_app/lib/admin_app_web/controllers/product_controller.ex]: use path helpers for redirect URL

File: apps/admin_app/lib/admin_app_web/router.ex [hunk 4]
```diff
@@ -77,6 +77,23 @@ defmodule AdminAppWeb.Router do
     delete("/taxonomy/delete", TaxonomyController, :delete_taxonomy)
     resources("/taxonomy", TaxonomyController, except: [:update])
     post("/taxonomy/create", TaxonomyController, :create_taxonomy)
+
+    get("/products/:product_id/property", ProductController, :index_property)
```
Comment (SagarKarwande) [inline apps/admin_app/lib/admin_app_web/router.ex]: Can we use [Nested Resources](https://hexdocs.pm/phoenix/routing.html#nested-resources) here for routes?

Comment (anantanant2015) [reply apps/admin_app/lib/admin_app_web/router.ex]: No we have added all the product property assoc handlers in product controller itself

File: apps/admin_app/lib/admin_app_web/controllers/product_controller.ex (hunk 3 above)
Comment (anantanant2015) [reply apps/admin_app/lib/admin_app_web/controllers/product_controller.ex]: done

File: apps/admin_app/lib/admin_app_web/controllers/product_controller.ex (hunk 2 above)
Comment (anantanant2015) [reply apps/admin_app/lib/admin_app_web/controllers/product_controller.ex]: done

File: apps/admin_app/lib/admin_app_web/controllers/product_controller.ex (hunk 1 above)
Comment (anantanant2015) [reply apps/admin_app/lib/admin_app_web/controllers/product_controller.ex]: done

**Pull request merged**